node/pkg/aggregator/aggregator_test.go

```diff
@@ -209,3 +209,102 @@ func TestDeactivateAggregatorByAdmin(t *testing.T) {
 	assert.NotNil(t, aggregator)
 	assert.False(t, aggregator.isRunning)
 }
+
+func TestStartAppByAdmin(t *testing.T) {
+	ctx := context.Background()
+	cleanup, testItems, err := setup(ctx)
+	if err != nil {
+		t.Fatalf("error setting up test: %v", err)
+	}
+	defer cleanup()
+
+	testItems.app.subscribe(ctx)
+
+	err = testItems.app.initialize(ctx, *testItems.host, testItems.pubsub)
+	if err != nil {
+		t.Fatal("error initializing app")
+	}
+
+	_, err = tests.RawPostRequest(testItems.admin, "/api/v1/aggregator/start", nil)
+	if err != nil {
+		t.Fatalf("error starting app: %v", err)
+	}
+
+	time.Sleep(100 * time.Millisecond)
+	assert.Equal(t, true, testItems.app.Aggregators[testItems.tmpData.aggregator.ID].isRunning)
+}
+
+func TestStopAppByAdmin(t *testing.T) {
+	ctx := context.Background()
+	cleanup, testItems, err := setup(ctx)
+	if err != nil {
+		t.Fatalf("error setting up test: %v", err)
+	}
+	defer cleanup()
+
+	testItems.app.subscribe(ctx)
+
+	err = testItems.app.initialize(ctx, *testItems.host, testItems.pubsub)
+	if err != nil {
+		t.Fatal("error initializing app")
+	}
+
+	_, err = tests.RawPostRequest(testItems.admin, "/api/v1/aggregator/start", nil)
+	if err != nil {
+		t.Fatalf("error starting app: %v", err)
+	}
+	time.Sleep(100 * time.Millisecond)
+	assert.Equal(t, true, testItems.app.Aggregators[testItems.tmpData.aggregator.ID].isRunning)
+
+	_, err = tests.RawPostRequest(testItems.admin, "/api/v1/aggregator/stop", nil)
+	if err != nil {
+		t.Fatalf("error stopping app: %v", err)
+	}
+
+	time.Sleep(150 * time.Millisecond)
+	assert.Equal(t, false, testItems.app.Aggregators[testItems.tmpData.aggregator.ID].isRunning)
+}
+
+func TestRefreshAppByAdmin(t *testing.T) {
+	ctx := context.Background()
+	cleanup, testItems, err := setup(ctx)
+	if err != nil {
+		t.Fatalf("error setting up test: %v", err)
+	}
+	defer cleanup()
+
+	testItems.app.subscribe(ctx)
+
+	err = testItems.app.initialize(ctx, *testItems.host, testItems.pubsub)
+	if err != nil {
+		t.Fatal("error initializing app")
+	}
+
+	lengthBefore := len(testItems.app.Aggregators)
+
+	_, err = tests.RawPostRequest(testItems.admin, "/api/v1/aggregator/start", nil)
+	if err != nil {
+		t.Fatalf("error starting app: %v", err)
+	}
+
+	time.Sleep(100 * time.Millisecond)
```

Comment:
`RawPostRequest` calls `raReq` which makes regular http POST request and then returns result body. It looks like the tests where failing without adding `time.Sleep`. Can you guess why it was happening?

Reply:
it's because these endpoints `start`, `stop`, and `refresh` works through message bus. there is slight delay between trigger and execution. it's similar to time.sleep you saw last time

Reply:
Aha! Isn't it possible to "slow down" the response so it will return only after the "task" was actually accomplished?

Reply:
yes it's possible I'll make an update